lib/iris/experimental/regrid.py

```diff
@@ -958,16 +1043,9 @@ def _regrid_indices(cells, depth, points):
     # target cube cell.
 
     # Determine the valid indices and their offsets in M x N space.
-    if ma.isMaskedArray(src_cube.data):
-        # Calculate the valid M offsets, accounting for the source cube mask.
-        mask = ~src_cube.data.mask.flatten()
-        cols = np.where((y_indices >= 0) & (y_indices < ty_depth) &
-                        (x_indices >= 0) & (x_indices < tx_depth) &
-                        mask)[0]
-    else:
-        # Calculate the valid M offsets.
-        cols = np.where((y_indices >= 0) & (y_indices < ty_depth) &
-                        (x_indices >= 0) & (x_indices < tx_depth))[0]
+    # Calculate the valid M offsets.
```

Comment:
Because it combined the which-src-in-what-tgt-cell calculation, which depends only on the grid, with the source-data-missing-points, which can vary between calls.
So the sparse matrix computed like this is specific to the source field, and can't be re-used on another 2d field.
The code in the "second part" (__perform) routine now handles the effects of the source .